autoendpoint/src/routers/fcm/client.rs

```diff
@@ -138,18 +154,24 @@ impl FcmClient {
 
         // Handle error
         let status = response.status();
-        if status.is_client_error() || status.is_server_error() {
-            let data: FcmResponse = response
-                .json()
-                .await
-                .map_err(FcmError::DeserializeResponse)?;
-
-            return Err(match (status, data.error) {
-                (StatusCode::UNAUTHORIZED, _) => RouterError::Authentication,
+        let data: GcmResponse = response
+            .json()
+            .await
+            .map_err(FcmError::DeserializeResponse)?;
+        if status.is_client_error() || status.is_server_error() || data.failure > 0 {
+            return Err(match (status, &data.results[0].error) {
+                (StatusCode::UNAUTHORIZED, _) => RouterError::GCMAuthentication,
                 (StatusCode::NOT_FOUND, _) => RouterError::NotFound,
```

Comment:
I'm not sure GCM ever sends us `NOT_FOUND`? Totally fine keeping this here but might warrant a comment that we're not sure it's used.

Reply:
I have no idea if we get a `NOT_FOUND`, that's why I'm fine keeping it here.